### PR TITLE
ch-workflow-superLinter

### PR DESCRIPTION
**Pull request by 911samuel**
---

**What does this PR do?**
This PR configures GitHub workflow for linting using Super-Linter.

**Description of Task to be completed**
The task involves configuring Super-Linter for GitHub workflows to ensure code quality and consistency across the project. The Super-Linter badge is added to the README to showcase the integration.

**How should this be manually tested?**
To test this PR manually, follow these steps:
1. Pull the changes from this PR branch into your local environment.
2. Open your project's README file.
3. Verify that the Super-Linter badge has been added to the README with the correct link.
4. Ensure that the GitHub workflow file for linting is correctly configured and includes Super-Linter.
5. Make some changes to your code files to introduce linting errors or inconsistencies.
6. Push the changes to your branch.
7. Wait for the GitHub workflow to run.
8. Check the workflow results to ensure that linting errors are caught and reported correctly.
9. Verify that the Super-Linter badge is updated accordingly based on the workflow status.

**Any background context you want to provide?**
Configuring Super-Linter for GitHub workflows ensures code quality and consistency across the project. Linting helps catch errors, enforce coding standards, and maintain a clean codebase.

**Questions:**
[If you have any questions or need clarifications, list them here]

---

### CODE REVIEW

**Comment by 911samuel** [comment]
Super-Linter status with a badge in  README:

[Super-Linter](https://github.com/marketplace/actions/super-linter)

**Review by fredshema — APPROVED** [review]

**Review by Latiah — APPROVED** [review]

File: .github/workflows/linter.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+on: 
+  push: null
+  pull_request: null
+
+permissions: { }
+
+jobs:
+  build:
+    name: Lint
+    runs-on: ubuntu-latest
```
Comment (Emmyfrank) [inline .github/workflows/linter.yml]: only runs in ubuntu?

**Review by Emmyfrank — APPROVED** [review]

File: .github/workflows/linter.yml (hunk 1 above)
Comment (911samuel) [reply .github/workflows/linter.yml]: thank youfor honest feedback i updated the lint now it is able to run on across all platform and also i added the functionality  to lint only newly pushed changes

**Comment by Adadie** [comment]
Consider also adding prettier to ensure the same formatting rules

**Comment by 911samuel** [comment]
Thank you for valuable feedback, I updated the workflow to allow prittier code formater

**Pull request merged**